packages/components/addon/components/hds/table/th-selectable.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { guidFor } from '@ember/object/internals';
+
+export default class HdsTableThSelectableComponent extends Component {
+  /**
+   * Generate a unique ID for the Checkbox
+   * @return {string}
+   */
+  checkboxId = 'checkbox-' + guidFor(this);
+
+  @action
+  onChange(event) {
+    let { onChange } = this.args;
+    if (typeof onChange === 'function') {
+      let { scope } = this.args;
+      if (scope === 'col') {
```

Comment:
as mentioned above, we could differentiate the function provided to the `@onChange` argument and avoid the need for these conditionals